### ISSUE TITLE
Question: Custom Express.js server cause MIME type ('text/html') issue

### ISSUE THREAD

**Issue by JunyuanZheng** [zeit/next.js#3305]
I am building my backend server based on the `next.js/custome-express example`, however, when I load a page I always get this error:

```
Refused to execute script from 'https://localhost:8081/_next/1511072683838/manifest.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.
```
I noticed in `dev` the  `1511072683838` is like a random number, however, in the `production` environment, it is an `uuid`. I am curious how can I use this number in my `express` server route, and is it safe to discard the number? e.g, having a route like:

```
app.use('/_next/:id/', constants.corsOptions, express.static(path.join(root, 'server', 'src', 'next', '.next')));
```

If I use this method and throw the `:id` parameter away will there be any issue (e.g security)? Or should I use this number as `nonce` in my `pages`, if that's the case how can I set the `nonce`.

Thanks!

- [x] I have searched the [issues](https://github.com/zeit/next.js/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior
As above.

## Current Behavior
As above.

## Context
Not sure if this will have any security issue.

## Your Environment
| Tech    | Version |
|---------|---------|
| next    |    4.1.4  |
| node    |    9.1     |
| OS      |    Docker Linux Alpine     |
| browser |    Chrome V63     |

**Comment by atuttle** [comment]
I started experiencing this yesterday too.

It turns out I introduced the problem with my "you're not logged in" redirect:

```js
server.use((req, res, next) => {
	if (!('user' in req.session)) {
		if (req.originalUrl === '/login') {
			return next();
		} else {
			return res.redirect('/login');
		}
	} else {
		return next();
	}
});
```

The problem is that the user is being redirected to `/login` and then the various requests for assets on that screen (including the JS needed by next.js) also get redirected to `/login`; so when the browser was expecting some executable `application/javascript` code, it received some `text/html` as the content of my `/login` screen.

The trick is to add exclusions for the other assets that it will be requesting so that they don't get redirected to `/login` too.

Hope that helps you!
